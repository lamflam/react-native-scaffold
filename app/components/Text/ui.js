import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Text as RNText } from 'react-native';
import { styled } from './styles';

export class Text extends PureComponent {
    static propTypes = {
        styles: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.number, PropTypes.object])).isRequired,
        style: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
        bold: PropTypes.bool
    };

    static defaultProps = {
        bold: false,
        style: null
    };

    render() {
        const { bold, children, styles, style, ...rest } = this.props;
        return (
            <RNText style={[bold ? styles.bold : styles.text, style]} {...rest}>
                {children}
            </RNText>
        );
    }
}

export const TextUI = styled(Text);
