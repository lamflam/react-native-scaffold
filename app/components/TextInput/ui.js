import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { TextInput as RNTextInput } from 'react-native';
import { styled } from './styles';

class TextInput extends PureComponent {
    static propTypes = {
        styles: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.number, PropTypes.object])).isRequired,
        style: PropTypes.oneOfType([PropTypes.array, PropTypes.number, PropTypes.object])
    };

    static defaultProps = {
        style: null
    };

    render() {
        const { styles, style, ...rest } = this.props;
        return <RNTextInput style={[styles.input, style]} {...rest} />;
    }
}

export const TextInputUI = styled(TextInput);
