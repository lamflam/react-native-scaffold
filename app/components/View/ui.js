import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { View as RNView } from 'react-native';
import { styled } from './styles';

export class View extends PureComponent {
    static propTypes = {
        styles: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.number, PropTypes.object])).isRequired,
        style: PropTypes.oneOfType([PropTypes.number, PropTypes.object])
    };

    static defaultProps = {
        style: null
    };

    render() {
        const { children, styles, style, ...rest } = this.props;
        return (
            <RNView style={[styles.view, style]} {...rest}>
                {children}
            </RNView>
        );
    }
}

export const ViewUI = styled(View);
