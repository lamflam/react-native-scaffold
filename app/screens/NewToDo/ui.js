import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { View, Text, TextInput } from 'app/components';
import { styled } from './styles';

class NewToDo extends PureComponent {
    static propTypes = {
        styles: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.number, PropTypes.object])).isRequired,
        updateTodo: PropTypes.func,
        todo: PropTypes.string,
        addToDo: PropTypes.func
    };

    static defaultProps = {
        todo: '',
        updateTodo: () => {},
        addToDo: () => {}
    };

    render() {
        const { addToDo, updateTodo, styles, todo } = this.props;
        return (
            <View style={styles.container}>
                <View style={styles.innerContainer}>
                    <TextInput placeholder="Add todo" style={styles.todoInput} onChangeText={updateTodo} value={todo} />
                </View>
                <Text style={styles.button} onPress={addToDo}>
                    Save
                </Text>
            </View>
        );
    }
}

export const NewToDoUI = styled(NewToDo);
