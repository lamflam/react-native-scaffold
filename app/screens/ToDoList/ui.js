/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { PureComponent } from 'react';
import { Text, View } from 'react-native';
import { styles } from './styles';

export class ToDoListUI extends PureComponent {
    render() {
        return (
            <View style={styles.container}>
                <Text style={styles.welcome}>ToDoList to React Native!</Text>
                <Text style={styles.instructions}>Build React Native app</Text>
                <Text style={styles.instructions}>Profit!</Text>
            </View>
        );
    }
}
