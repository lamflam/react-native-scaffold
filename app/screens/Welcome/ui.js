import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Text, View } from 'react-native';
import { styled } from './styles';

export class Welcome extends PureComponent {
    static propTypes = {
        styles: PropTypes.objectOf(PropTypes.oneOfType([PropTypes.number, PropTypes.object])).isRequired,
        toNextScreen: PropTypes.func.isRequired
    };

    render() {
        const { styles, toNextScreen } = this.props;
        return (
            <View style={styles.container}>
                <Text style={styles.welcome}>Welcome to React Native Scaffolding!</Text>
                <Text onPress={toNextScreen} style={styles.instructions}>
                    Go to my To Do List
                </Text>
            </View>
        );
    }
}

export const WelcomeUI = styled(Welcome);
