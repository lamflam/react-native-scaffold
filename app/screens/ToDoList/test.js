import React from 'react';
import { testSnapshot } from 'app/utils/tests';
import { ToDoListUI } from './ui';

describe('ToDoListUI UI', () => {
    it('matches snapshot', () => {
        testSnapshot(<ToDoListUI />);
    });
});
