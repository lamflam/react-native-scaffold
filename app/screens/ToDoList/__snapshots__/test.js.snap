// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoListUI UI matches snapshot 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    ToDoList to React Native!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    Build React Native app
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    Profit!
  </Text>
</Component>
`;
